### ISSUE TITLE
[Feature request] timers widget

### ISSUE THREAD

**Issue by chr15m** [SimpleMobileTools/Simple-Clock#414]
All in the subject. Would be great to be able to put timers on the home screen for fast access to start a countdown. Use case: 3 minute timer for brewing tea.

Thank you for your amazing work on this suite of apps!

**Comment by ZineeEddine** [comment]
Hello, I am currently studying computer science at university and i like mobile app development.
First of all congratulations for this very good project.

I want to contribute to this Timer Widget feature.

I did some test but the widget has a problem when I want to put an editTextNumber inside, here are some screenshots:

![Screenshot_20230201_224748](https://user-images.githubusercontent.com/92603736/216196427-e249d8aa-46db-4d22-90cb-8869dafba2aa.png)

![Screenshot_20230201_225158](https://user-images.githubusercontent.com/92603736/216197059-04f287f9-7e9d-4864-a8ee-69f6c99f228a.png)

I know how to make a timer with 3 minutes in a widget with a button to activate and reset it, but I can't make a timer where the user chooses the time.

Can you tell me the procedure to follow to code a timer in a widget with which the user can choose the time he wants?

**Comment by chr15m** [comment]
Thank you for progressing this issue. It's possible to create the timers in Simple Clock already:

![image](https://user-images.githubusercontent.com/67130/216204221-5b89df33-c900-4a9b-8bf8-57f43b03a057.png)

It would make sense to put a button or menu option to add one of these existing timers as a widget to the home screen. The other option would be when creating the widget to read Simple Clock's database of existing timers and allow the user to select one. That way you don't have to build a new UI to do it.

**Comment by ZineeEddine** [comment]
thank you for your answer.

That's what I had considered but for user experience, I thought it would be better to configure the widget directly and not in the application. 

 It's ok I'll try, to do it from the application or with the database.

**Comment by ZineeEddine** [comment]
I forked the project, I was working on an option when choosing in the EditTimerDialog :

I added a BUTTON_NEUTRAL, which is called "Add Widget" (to be translated into other languages), so that the user can choose whether to save the timer in the list of timers by pressing OK or in the widget by pressing ADD WIDGET.

I haven't committed to my new branch yet but I think this may be a good start for a new Timer Widget feature.

Here is a screens video to illustrate the example :

https://user-images.githubusercontent.com/92603736/216413163-2f679b24-e739-45ad-8799-588fcb72d852.mp4

**Comment by chr15m** [comment]
@ZineeEddine :100: love it, hopefully your PR will get merged once you're ready.

**Comment by tibbi** [comment]
widgets should be added the casual way people are used to, such buttons shouldnt be at dialogs.

**Comment by chr15m** [comment]
@tibbi do you mean via e.g. home screen and then choose timers from the Clock db?

**Comment by tibbi** [comment]
sure, thats how it should be done

**Comment by ZineeEddine** [comment]
@tibbi 

can you give me more details please

do I have to include in the widget, a LISTVIEW with all the timers in the database?

does the user have to scroll the widget if he wants to choose a timer?

**Comment by tibbi** [comment]
no, lets show just 1 timer per widget, maybe show a timer picker at the config screen, where you are picking colors. Check Simple Notes, there is a note picker at the widget config screen, if you have multiple notes created. I imagine it could work the same way.
